src/msg/async/AsyncMessenger.cc

```diff
@@ -357,9 +357,11 @@ int AsyncMessenger::shutdown()
   return 0;
 }
 
-int AsyncMessenger::bind(const entity_addr_t &bind_addr)
+int AsyncMessenger::bind(const entity_addr_t &bind_addr,
+                         std::optional<entity_addrvec_t> _public_addrs)
```

Comment:
This is one of the reasons why members should be prefixed with `m_` -- unfortunately not followed in `AsyncMessenger`.  I'd suggest renaming the member to e.g. `saved_public_addrs` to avoid the underscore ugliness.